### PR TITLE
[pipelining] Add pipeline stage test

### PR DESCRIPTION
**Pull request by kwen2501**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #126735
* __->__ #126721
* #126812

Test tracer's and manual's stage creation by using a basic schedule (GPipe).

(Migrated from https://github.com/pytorch/PiPPy/blob/main/test/test_pipeline_stage.py)

Test command:
```
$ python test_stage.py
```

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

**Review by wconstab — APPROVED** [review]
thanks!

File: test/distributed/pipelining/test_stage.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+
+d_hid = 512
+batch_size = 256
+chunks = 4
+
+torch.manual_seed(0)
+
+
+class StageTest(MultiProcContinousTest):
```
Comment (H-Huang) [inline test/distributed/pipelining/test_stage.py:38]: Just curious, what does `MultiProcContinousTest` do? How is it different than `MultiProcessTestCase`

**Review by H-Huang — APPROVED** [review]
Nice!

**Pull request closed without merging**